### PR TITLE
Updated vendored react-native-webview to v13.2.2

### PR DESCRIPTION
**Pull request by aleqsio**
# Why

SDK49

# How

It needed some changes to the legacy vendoring code. The new version stores native files in separate folders for new and old arch and uses gradle files to combine two folders.

# Test Plan

Tested manually on both platforms.

# Checklist

- [ ] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [ ] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `npx expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

File: tools/src/vendoring/config/expoGoConfig.ts [hunk 1]
```diff
@@ -287,33 +287,35 @@ const config: VendoringTargetConfig = {
 `,
```
Comment (aleqsio) [inline tools/src/vendoring/config/expoGoConfig.ts:287]: @Kudo here's some changes that I'd appreciate a review on

File: tools/src/vendoring/legacy.ts [hunk 2]
```diff
@@ -590,6 +590,24 @@ const vendoredModulesConfig: { [key: string]: VendoredModuleConfig } = {
         sourceAndroidPackage: 'com.reactnativecommunity.webview',
         targetAndroidPackage: 'versioned.host.exp.exponent.modules.api.components.webview',
       },
+      {
```
Comment (aleqsio) [inline tools/src/vendoring/legacy.ts:593]: @Kudo same here :)

File: tools/src/vendoring/config/expoGoConfig.ts [hunk 3]
```diff
@@ -287,33 +287,35 @@ const config: VendoringTargetConfig = {
...
-              paths: 'RNCWebViewManager.m',
+              paths: 'RNCWebViewManager.mm',
               find: /@implementation RNCWebViewManager\s*{/,
               replaceWith: '$&\n  NSString *_scopeKey;',
             },
             {
-              paths: 'RNCWebViewManager.m',
+              paths: 'RNCWebViewManager.mm',
               find: '*webView = [RNCWebView new];',
               replaceWith: '*webView = [RNCWebView new];\n  webView.scopeKey = _scopeKey;',
```
[the remark this answers is not in the document]
Comment (aleqsio) [reply tools/src/vendoring/config/expoGoConfig.ts:307]: Sorry, missed that, will double check to see if all transforms got updated.

Perhaps we should make the vendoring process error out if a transform never got applied?

**Comment by aleqsio** [comment]
Thank you for the catch @Kudo, I've checked that all code transformations for iOS get applied correctly.

**Review by Kudo — APPROVED** [review]

**Review by tsapeta — APPROVED** [review]
I'm impressed how this package migrates to Kotlin and also back to Java at the same time. `RNCWebViewPackage` has been migrated from Kotlin to Java, there are a few new files in Java and two new files in Kotlin 😂

**Pull request merged**